pylint/checkers/strings.py

```diff
@@ -873,19 +871,15 @@ def process_non_raw_string_token(
         # sequence <https://docs.python.org/reference/lexical_analysis.html>
         #
         index = 0
-        while True:
-            index = string_body.find("\\", index)
-            if index == -1:
-                break
+        while (index := string_body.find("\\", index)) != -1:
```

Comment:
Here's a nice case where using an assignment expression makes the logic dramatically simpler.